### DISCUSSION TITLE
Can i change verify flow?

### DISCUSSION THREAD

**Discussion by GoodGoodJM** [ory/kratos#3447] [Q&A] [answered]
hello i am just learning and using duck ory-kratos to use our services.
Ory is very beautiful project! I fallen in love.

Our service requires email verification before user registration. We need to allow users to sign up only when they have verified their email.

I found [verify email step](https://www.ory.sh/docs/kratos/self-hosted/account-activation-email-verification) docs, but it seems to be verify email after sign up.

**Comment by vinckr** [comment, marked as answer]
Hello @GoodGoodJM  
I'm glad to hear that you're enjoying using Ory Kratos!

Ory Kratos does indeed support email verification, but it typically happens after the user has signed up. When a user signs up, if verification is enabled, Ory Kratos starts the verification process automatically. Users can also verify their addresses manually. The verification flow is supported in both browsers and API clients.

If you want to allow only the users with a verified email to sign in, you can follow these steps:

Either:
Go to Authentication → Email Verification in the [Ory Console](https://console.ory.sh/projects/current/verification) and enable Require Verified Address for Login.

Or:
```
Download the Ory Identities config from your project and save it to a file:

## List all available projects  
ory list projects  
  
## Get config  
ory get identity-config {project-id} --format yaml > identity-config.yaml
Add the configuration for the verification flow

selfservice:  
  flows:  
  login:  
+ after:  
+ hooks:  
+ - hook: require_verified_address

Update the Ory Identities configuration using the file you worked with:
ory update identity-config {project-id} --file updated_config.yaml
```
Note: This is for Ory Network, if you are self-hosting edit the configuration directly.

For more information see the [hooks configuration](https://www.ory.sh/docs/kratos/hooks/configure-hooks) documentation.

Please note that this configuration will require users to verify their email address before they can log in, not before they can sign up. 

If you want to enforce email verification before sign up, you might need to implement a custom solution using Ory Actions. See this example of [Ory Action to check IP addresses against VPN and Fraud detection services](https://github.com/ory/examples/tree/master/ory-actions/vpncheck-py#ory-action-to-check-ip-addresses-against-vpn-and-fraud-detection-services).
